cmd/dendrite-appservice-server/main.go

```diff
@@ -35,5 +35,9 @@ func main() {
 		base, accountDB, deviceDB, federation, alias, query, cache,
 	)
 
-	base.SetupAndServeHTTP(string(base.Cfg.Listen.FederationSender))
+	if base.Cfg.Bind.AppServiceAPI != "" {
```

Comment:
It might be cleaner just to change `SetupAndServeHTTP` to take both addresses and have the fallback logic inside of it instead?

Reply:
Pushed a change, can you make sure its alright?